salt/modules/yumpkg.py

```diff
@@ -2100,6 +2101,24 @@ def remove(name=None, pkgs=None, **kwargs):  # pylint: disable=W0613
 
     old = list_pkgs()
     targets = []
+
+    # Loop through pkg_params looking for any
+    # which contains a wildcard and get the
+    # real package names from the packages
+    # which are currently installed.
+    pkg_matches = {}
+    for pkg_param in copy.deepcopy(pkg_params):
```

Comment:
```suggestion
    for pkg_param in list(pkg_params):
```
A copy of the keys should be enough?